org.elasticsearch.index.translog.TranslogTests#testStats failing on 6.8

Observed a failure on branch 6.8 for `TranslogTests` in `#testStats`:

```
  | FAILURE 0.02s J15 \| TranslogTests.testStats <<< FAILURES! |  
  | > Throwable #1: java.lang.AssertionError: |  
  | > Expected: a value greater than <1L> |  
  | >      but: <1L> was equal to <1L> |  
  | > 	at __randomizedtesting.SeedInfo.seed([6B2A44E9E0939A49:F1AB1D56884F546A]:0) |  
  | > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20) |  
  | > 	at org.elasticsearch.index.translog.TranslogTests.testStats(TranslogTests.java:439) |  
  | > 	at java.lang.Thread.run(Thread.java:748)
```
Reproduction line:

`./gradlew ':server:unitTest' -Dtests.seed=6B2A44E9E0939A49 -Dtests.class=org.elasticsearch.index.translog.TranslogTests -Dtests.method="testStats" -Dtests.security.manager=true -Dtests.locale=ru -Dtests.timezone=SystemV/MST7MDT -Dcompiler.java=12 -Druntime.java=8`

Unfortunately I wasn't able to reproduce after 50 iterations on the same operating system.

CI link: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+matrix-java-periodic/ES_RUNTIME_JAVA=adoptopenjdk8,nodes=general-purpose/622/console / https://gradle-enterprise.elastic.co/s/6xn7qjvruu5x6

Interestingly enough this is the first time if fails on `6.8`, however, there were a number of failures on `master` in February that got fixed by #51905.

@dnhatn :

The specific line in 6.8 asserts >1 in https://github.com/elastic/elasticsearch/blob/fe9394ca81417f02ea5bd3e9ed02029cf394ba58/server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java#L439

however, in #51905, we started asserting >0 https://github.com/elastic/elasticsearch/blob/6d28596ead7f95182b8bd112c3a314d540728126/server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java#L420

Is there a reason for this difference?

--- a/server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java
+++ b/server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java
@@ -419,6 +419,8 @@ public class TranslogTests extends ESTestCase {
     }
 
     public void testStats() throws IOException {
+        assertThat(translog.getCurrent().getLastModifiedTime(), lessThanOrEqualTo(System.currentTimeMillis()));
+        long lastModifiedAge = System.currentTimeMillis() - translog.getCurrent().getLastModifiedTime();
         // self control cleaning for test
         translog.getDeletionPolicy().setRetentionSizeInBytes(1024 * 1024);
         translog.getDeletionPolicy().setRetentionAgeInMillis(3600 * 1000);
@@ -436,7 +438,7 @@ public class TranslogTests extends ESTestCase {
             assertThat(stats.getTranslogSizeInBytes(), equalTo(164L));
             assertThat(stats.getUncommittedOperations(), equalTo(1));
             assertThat(stats.getUncommittedSizeInBytes(), equalTo(164L));
-            assertThat(stats.getEarliestLastModifiedAge(), greaterThan(1L));
+            assertThat(stats.getEarliestLastModifiedAge(), greaterThanOrEqualTo(lastModifiedAge));
         }
 
         translog.add(new Translog.Delete("test", "2", 1, primaryTerm.get(), newUid("2")));
@@ -446,7 +448,7 @@ public class TranslogTests extends ESTestCase {
             assertThat(stats.getTranslogSizeInBytes(), equalTo(213L));
             assertThat(stats.getUncommittedOperations(), equalTo(2));
             assertThat(stats.getUncommittedSizeInBytes(), equalTo(213L));
-            assertThat(stats.getEarliestLastModifiedAge(), greaterThan(1L));
+            assertThat(stats.getEarliestLastModifiedAge(), greaterThanOrEqualTo(lastModifiedAge));
         }
 
         translog.add(new Translog.Delete("test", "3", 2, primaryTerm.get(), newUid("3")));
@@ -456,7 +458,7 @@ public class TranslogTests extends ESTestCase {
             assertThat(stats.getTranslogSizeInBytes(), equalTo(262L));
             assertThat(stats.getUncommittedOperations(), equalTo(3));
             assertThat(stats.getUncommittedSizeInBytes(), equalTo(262L));
-            assertThat(stats.getEarliestLastModifiedAge(), greaterThan(1L));
+            assertThat(stats.getEarliestLastModifiedAge(), greaterThanOrEqualTo(lastModifiedAge));
         }
 
         translog.add(new Translog.NoOp(3, 1, randomAlphaOfLength(16)));
@@ -466,7 +468,7 @@ public class TranslogTests extends ESTestCase {
             assertThat(stats.getTranslogSizeInBytes(), equalTo(304L));
             assertThat(stats.getUncommittedOperations(), equalTo(4));
             assertThat(stats.getUncommittedSizeInBytes(), equalTo(304L));
-            assertThat(stats.getEarliestLastModifiedAge(), greaterThan(1L));
+            assertThat(stats.getEarliestLastModifiedAge(), greaterThanOrEqualTo(lastModifiedAge));
         }
 
         final long expectedSizeInBytes = 359L;
@@ -477,7 +479,7 @@ public class TranslogTests extends ESTestCase {
             assertThat(stats.getTranslogSizeInBytes(), equalTo(expectedSizeInBytes));
             assertThat(stats.getUncommittedOperations(), equalTo(4));
             assertThat(stats.getUncommittedSizeInBytes(), equalTo(expectedSizeInBytes));
-            assertThat(stats.getEarliestLastModifiedAge(), greaterThan(1L));
+            assertThat(stats.getEarliestLastModifiedAge(), greaterThanOrEqualTo(lastModifiedAge));
         }
 
         {
@@ -507,7 +509,7 @@ public class TranslogTests extends ESTestCase {
             assertThat(stats.getTranslogSizeInBytes(), equalTo(expectedSizeInBytes));
             assertThat(stats.getUncommittedOperations(), equalTo(0));
             assertThat(stats.getUncommittedSizeInBytes(), equalTo(firstOperationPosition));
-            assertThat(stats.getEarliestLastModifiedAge(), greaterThan(1L));
+            assertThat(stats.getEarliestLastModifiedAge(), greaterThanOrEqualTo(lastModifiedAge));
         }
     }
 

